File: 001_two_sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hashMap = {}

        for i in range(len(nums)):
            wanted = target - nums[i]
            if wanted in hashMap:
                return [i, hashMap[wanted]]
            else:
                hashMap[nums[i]] = i

File: 001_two_sum/two_sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> hashMap = new HashMap <Integer, Integer>();
        
        for (int i = 0; i < nums.length; i++){
            int wanted = target - nums[i];
            if (hashMap.get(wanted) != null){
                return new int[] {i, hashMap.get(wanted)};
            }else{
                hashMap.put(nums[i], i);
            }
        }
        
        return null;
    }
}